front/lib/api/assistant/actions/search_labels.ts

```diff
@@ -0,0 +1,277 @@
+  async renderForMultiActionsModel(): Promise<FunctionMessageTypeModel> {
+    let content = "";
+    if (this.output === null) {
+      content = "No tags found.";
+    } else {
+      content = "Available tags:\n";
+      for (const tag of this.output.tags) {
+        content += `- ${tag.tag} (available in ${tag.data_sources.length} data sources)\n`;
+      }
+    }
+
+    return {
+      role: "function",
+      name: this.functionCallName,
+      function_call_id: this.functionCallId ?? `call_${this.id.toString()}`,
+      content,
+    };
+  }
+}
+
+/**
+ * Params generation.
+ */
+
+export class SearchLabelsConfigurationServerRunner extends BaseActionConfigurationServerRunner<SearchLabelsConfigurationType> {
+  async buildSpecification(
+    auth: Authenticator,
+    {
+      description,
+      name,
+    }: {
+      description: string | null;
+      name: string;
+    }
+  ): Promise<Result<AgentActionSpecification, Error>> {
+    return new Ok({
+      name,
+      description:
+        description ||
+        `Find exact matching labels before using them in the tool ${this.actionConfiguration.parentTool}. ` +
+          "Restricting or excluding content succeeds only with existing labels. " +
+          "Searching without verifying labels first typically returns no results.",
+      inputs: [
+        {
+          name: "searchText",
+          description:
+            "The text to search for in existing labels. " +
+            "The returned labels can be used in tagsIn/tagsNot parameters to restrict or exclude content " +
+            "based on the user request and conversation context.",
+          type: "string",
+        },
+      ],
+    });
+  }
+
+  async *run(
+    auth: Authenticator,
+    { agentMessage, rawInputs, functionCallId, step }: BaseActionRunParams
+  ): AsyncGenerator<
+    SearchLabelsParamsEvent | SearchLabelsSuccessEvent | SearchLabelsErrorEvent,
+    void
+  > {
+    const { actionConfiguration } = this;
+
+    const { searchText } = rawInputs;
+
+    if (!isString(searchText)) {
+      yield {
+        type: "search_labels_error",
+        created: Date.now(),
+        configurationId: actionConfiguration.sId,
+        messageId: agentMessage.sId,
+        error: {
+          code: "search_labels_parameters_generation_error",
+          message:
+            "The searchText parameter is required and must be a valid string.",
+        },
+      };
+
+      return;
+    }
+
+    // Fetch data source views to get dustAPIDataSourceIds.
+    const dataSourceViews = await DataSourceViewResource.fetchByIds(
+      auth,
+      this.actionConfiguration.dataSourceViewIds
+    );
+
+    // Ensure the auth can read all the data source views.
+    assert(
+      dataSourceViews.some((dsv) => dsv.canRead(auth)),
+      "Unauthorized attempt to read data source view in `search_labels` action"
+    );
+
+    // Get the dustAPIDataSourceIds from the data source views.
+    const dustAPIDataSourceIds = dataSourceViews.map(
```

Comment:
Here we're definitely leaking potentially critical data (the labels of files you don't have access to ?)

Reply:
Thanks!